src/lib/WIFI/devWIFI.cpp

```diff
@@ -796,6 +801,238 @@ static void WebUploadForceUpdateHandler(AsyncWebServerRequest *request) {
   }
 }
 
+#if defined(RADIO_LR1121)
+static size_t expectedFilesize;
+static SX12XX_Radio_Number_t updatingRadio;
+extern LR1121Hal hal;
+static uint8_t packet[256];
+static size_t left_over;
+
+static void writeLR1121Bytes(uint8_t *data, uint32_t sector_size) {
+    uint8_t header[6 + 256] = {
+        (uint8_t)(LR11XX_BL_WRITE_FLASH_ENCRYPTED_OC >> 8),
+        (uint8_t)(LR11XX_BL_WRITE_FLASH_ENCRYPTED_OC),
+        (uint8_t)(totalSize >> 24),
+        (uint8_t)(totalSize >> 16),
+        (uint8_t)(totalSize >> 8),
+        (uint8_t)(totalSize)
+    };
+    if (left_over)
+    {
+        DBGLN("left %x", left_over);
+        memcpy(header + 6, packet, left_over);
+    }
+    if (data != NULL)
+    {
+        DBGLN("new %x", sector_size - left_over);
+        memcpy(header + 6 + left_over, data, sector_size - left_over);
+    }
+    DBGLN("flashing %x at %x", sector_size, totalSize);
+
+    // Have to do this the OLD way, so we can pump out more than 64 bytes in one message
+    digitalWrite(updatingRadio == SX12XX_Radio_1 ? GPIO_PIN_NSS : GPIO_PIN_NSS_2, LOW);
+    SPIEx.transferBytes(header, NULL, 6 + sector_size);
+    digitalWrite(updatingRadio == SX12XX_Radio_1 ? GPIO_PIN_NSS : GPIO_PIN_NSS_2, HIGH);
+
+    while (digitalRead(updatingRadio == SX12XX_Radio_1 ? GPIO_PIN_BUSY : GPIO_PIN_BUSY_2) == HIGH)
+    {
+        delay(1);
+    }
+    DBGLN("flashed");
+}
+
+static void WebUploadLR1121ResponseHandler(AsyncWebServerRequest *request) {
+    // Complete upload and set error flag
+    bool uploadError = false;
+    uint8_t param = 0;
+    writeLR1121Bytes(NULL, left_over);
```

Comment:
should probably pass 0 here to not mix the argument convention. `writeLR1121Bytes()` always will internally add `left_over` to the data passed, so passing it here implies that the pointer _also_ contains data, but the pointer is NULL (should also be `nullptr`) so this is doubling up.